c7n/registry.py

```diff
@@ -45,19 +45,32 @@ class PluginRegistry(object):
 
     """
 
+    EVENT_REGISTER = 0
+    EVENT_FINAL = 1
+    EVENTS = (EVENT_REGISTER, EVENT_FINAL)
+
     def __init__(self, plugin_type):
         self.plugin_type = plugin_type
         self._factories = {}
+        self._subscribers = {x: [] for x in self.EVENTS}
+
+    def subscribe(self, event, func):
+        if event not in self.EVENTS:
+            raise ValueError('Invalid event')
+
+        self._subscribers[event].append(func)
 
     def register(self, name, klass=None):
         # invoked as function
         if klass:
+            self.notify(self.EVENT_REGISTER, name)
```

Comment:
i'd suggest modifying the param here to be k/class and do it a few lines laster.